js-pkg/packages/driftdb-react/src/index.tsx

```diff
@@ -315,6 +316,56 @@ export function useLatency(): number | null {
   return latency
 }
 
+type DataChannelMsg = { sender: string; value: any; lastSeen: number }
+type WebRtcOnMessage = (msg: DataChannelMsg) => void
+
+function useWebRtcBroadcastChannel(throttle = 0) {
+  const db = useDatabase()
+  const id = useUniqueClientId()
+  const WebRtcBroadcastChannelRef = React.useRef<SyncedWebRTCConnections>()
+  if (!WebRtcBroadcastChannelRef.current) {
+    WebRtcBroadcastChannelRef.current = new SyncedWebRTCConnections(db, id, throttle)
+  }
+  return {
+    send: (msg: string) => WebRtcBroadcastChannelRef.current!.send(msg),
+    setOnMessage: (onMessage: WebRtcOnMessage) =>
+      WebRtcBroadcastChannelRef.current!.setOnMessage(onMessage),
+    peers: [...WebRtcBroadcastChannelRef.current!.peers()]
+  }
+}
+
+/**
+ * A React hook that returns a map of the current presence of all clients in the current room.
+ * The client also passes its own value, which wil be included in the map for other clients
+ *  @param value The value that will be included in the map for other clients.
+ *  @param throttle The minimum interval between messages being sent.
+ *         NOTE: any messages sent in the interval will be dropped.
+ *  @returns A map of the current presence of all clients in the current room.
+ */
+export function useWebRtcPresence<T>(
+  value: T,
+  throttle = 0
+): Record<string, WrappedPresenceMessage<T>> {
+  const { send, setOnMessage, peers } = useWebRtcBroadcastChannel(throttle)
+  const rtcMap = useRef<Record<string, WrappedPresenceMessage<T>>>({})
+  for (const peer of Object.keys(rtcMap.current!)) {
+    if (!peers.includes(peer)) delete rtcMap.current[peer]
+  }
+  const [_, setMessageCount] = useState(0)
+  React.useEffect(() => {
+    send(JSON.stringify(value))
+  }, [value])
+  React.useEffect(() => {
+    let count = 0
+    setOnMessage((msg) => {
+      rtcMap.current[msg.sender] = { value: msg.value, lastSeen: msg.lastSeen }
+      setMessageCount(count++)
```

Comment:
It seems like what we really want is for changes to `rtcMap` to trigger rerenders. In that case, I'd suggest we use `useState` for the rtcMap instead of a ref. It looks like that's how `usePresence` works.

```ts
const [rtcPresence, setRtcPresence] = useState<Record<string, WrappedPresenceMessage<T>>>({})
// then we drop the messageCount stuff
React.useEffect(() => {
    setOnMessage((msg) => {
      rtcPresence[msg.sender] = { value: msg.value, lastSeen: msg.lastSeen }
      setRtcPresence(rtcPresence)
    })
})
```

Something like that?

Also, why can't we just pass this callback directly to the `useWebRtcBroadcastChannel()` instead of having it return a setter that we use right away? Did we already talk through this but I've just forgotten?